### PR TITLE
DRAFT: get consumer

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/jetstream/consumer.rs [hunk 1]
```diff
@@ -15,6 +15,99 @@ use std::time::Duration;
 use serde::{Deserialize, Serialize};
 use time::serde::rfc3339;
 
+use crate::Error;
+
+pub trait IntoConsumerConfig {
+    fn into_consumer(self) -> ConsumerConfig;
```
Comment (caspervonb) [inline async-nats/src/jetstream/consumer.rs:21]: Not sure if this should be ref self, ala `ToSocketAddrs` or an into trait like now.
But name should at-least end with `consumer_config`.

```suggestion
    fn into_consumer_config(self) -> ConsumerConfig;
```

Comment (caspervonb) [inline async-nats/src/jetstream/consumer.rs:21]: But probably returns Result<ConsumerConfig>, so we can validate?

**Review by caspervonb — APPROVED** [review]
Some nits we should sort out, but can punt on it.
LGTM.

File: async-nats/src/jetstream/consumer.rs (hunk 1 above)
Comment (Jarema) [reply async-nats/src/jetstream/consumer.rs:21]: There is no need to validate this way, as we're going Pull/Push specific config -> config.

File: async-nats/src/jetstream/consumer.rs [hunk 2]
```diff
@@ -15,6 +15,99 @@ use std::time::Duration;
...
+}
+
+impl<T: IntoConsumerConfig> Consumer<T> {
+    pub fn new(config: T) -> Self {
+        Self { config }
+    }
+}
+
+pub trait FromConsumer {
+    fn try_from_consumer_config(config: ConsumerConfig) -> Result<Self, Error>
```
[the remark this answers is not in the document]
Comment (Jarema) [reply async-nats/src/jetstream/consumer.rs:35]: Knowing issues with Send + Sync - we will follow up, as this is internal.

**Pull request merged**